lib/config/typescript-fast.js

```diff
@@ -0,0 +1,32 @@
+module.exports = {
+  overrides: [
+    {
+      parser: '@typescript-eslint/parser',
+      parserOptions: {
+        ecmaVersion: 2017,
+        sourceType: 'module',
+      },
+      files: ['*.ts', '*.tsx'],
+      rules: {
+        // Disallows awaiting a value that is not a Thenable
+        '@typescript-eslint/await-thenable': 'off',
```

Comment:
Is it possible to avoid using overrides? I feel like they cause a lot of problems in consumers because you need another override to override these rules. If only the parser bits are override, and the rules are not nested, does that break?

Reply:
I seem to recall we had something kinda like that before and it meant that typescript rules got applied to JS files, which caused problems because they didn't apply (a lint saying you should use `import` but you're writing a script to be ran in node where you only have access to `require()`).

The problems with needing overrides to trump overrides you mention have gone away in eslint 6 - they made a breaking change so that top-level rules now trump an override in an extended config - https://eslint.org/docs/user-guide/migrating-to-6.0.0#overrides-precedence. A parent config now always has precedence over everything in an extended config.

Reply:
I upgraded eslint in my tophat PR for this reason. https://github.com/Shopify/quilt/pull/1110/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R53

Reply:
Thank you both for the explanation ❤️